netshare/utils/field.py

```diff
@@ -32,42 +40,54 @@ def __init__(self, norm_option, min_x=None, max_x=None, dim_x=1,
     # Normalize x in [a, b]: x' = (b-a)(x-min x)/(max x - minx) + a
     def normalize(self, x):
         if x.shape[-1] != self.dim_x:
-            raise ValueError(f"Dimension is {x.shape[-1]}. "
-                             f"Expected dimension is {self.dim_x}")
+            raise ValueError(
+                f"Dimension is {x.shape[-1]}. Expected dimension is {self.dim_x}"
+            )
+        if self.log1p_norm:
+            x = np.log1p(x)
         # [0, 1] normalization
         if self.norm_option == Normalization.ZERO_ONE:
+            if self.max_x - self.min_x == 0:
+                raise Exception("Not enough data to proceed!")
             return np.asarray((x - self.min_x) / (self.max_x - self.min_x))
 
         # [-1, 1] normalization
         elif self.norm_option == Normalization.MINUSONE_ONE:
-            return np.asarray(2 * (x - self.min_x)
-                              / (self.max_x - self.min_x) - 1)
+            if self.max_x - self.min_x - 1 == 0:
```

Comment:
Why there is "-1"?

Reply:
You're right here, that was my miss. I tried to avoid `ZeroDivisionError`. Will be fixed in a sec!